components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -4724,6 +4735,46 @@ public API getAPIbyUUID(String uuid, String organization) throws APIManagementEx
         }
     }
 
+    @Override
+    public Map<String, Object> searchPaginatedAPIsByFQDN(String searchQuery, String tenantDomain, int start, int end) throws APIManagementException {
+        Map<String, Object> result = new HashMap<String, Object>();
+
+        Organization org = new Organization(organization);
+        String[] roles = APIUtil.getFilteredUserRoles("admin");
+        Map<String, Object> properties = APIUtil.getUserProperties("admin");
+        UserContext userCtx = new UserContext("admin", org, properties, roles);
+
+        try {
+            PublisherAPISearchResult searchAPIs = apiPersistenceInstance.searchAPIsForPublisher(org, searchQuery,
+                    start, end, userCtx, "createdTime", "desc");
+            if (log.isDebugEnabled()) {
+                log.debug("searched APIs for query : " + searchQuery + " :-->: " + searchAPIs.toString());
```

Comment:
Improve the log egz: Running query : + + to retrieve ....

Reply:
done